bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/XMLItem.java

```diff
@@ -639,20 +636,15 @@ public Object copy(Map<Object, Object> refs) {
                 clonedNode = ((OMElement) omNode).cloneOMElement();
                 break;
             case TEXT:
-                TextImpl text = new TextImpl();
-                text.setTextContent(((OMText) omNode).getText());
-                clonedNode = text;
+                clonedNode = (OMNode) XMLFactory.createXMLText(((OMText) omNode).getText()).value();
```

Comment:
`XMLFactory.createXMLText()` will return the `XMLItem` instance. We can directly return it.
Getting the value of it and creating a another `XMLItem` from it seems redundant and expensive.

Same for below two cases as well.

Reply:
There is this wrapping in OMDocument not before returning thing, is it ok skipping it?
```java 
        // adding the document element as parent, to get xpPaths work
        OMDocument doc = new OMDocumentImpl();
        doc.addChild(clonedNode);
        return new XMLItem(clonedNode);
```